logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -186,21 +177,14 @@ def cancel(event)
 
       def to_a
         events = []
-        each {|e| events << e}
+        @originals.each {|e| events << e unless e.cancelled?}
```

Comment:
This whole method can be 
```
def to_a
  @originals.select{|e| !e.cancelled?}
end
```
`select` will create a new array

Reply:
@guyboertje I "think" `select` will be slower here since `@originals` is a `java.util.HashSet` <= Ruby has to provide the select method via its wrapper logic for these collections.
I'd rather keep this and improve it another PR to simply take an array as argument for `to_a` so we don't have to recreate those `[]` constantly?

Reply:
hmm though the opposite could be true as well ... let me benchmark this :D sec

Reply:
hmm no measurable difference, but in theory `select` must be slower, it's the same code path plus some special case handling for size `1` and `2` which we never really make use of.
=> I'll leave it as is :)